src/libraries/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs

```diff
@@ -250,23 +246,14 @@ public override int Read(Span<byte> buffer)
             ThrowIfDisposed();
             if (!CanRead) throw new InvalidOperationException(SR.net_writeonlystream);
 
-            int bytesRead;
-            SocketError errorCode;
             try
             {
-                bytesRead = _streamSocket.Receive(buffer, SocketFlags.None, out errorCode);
+                return _streamSocket.Receive(buffer, SocketFlags.None);
             }
             catch (Exception exception) when (!(exception is OutOfMemoryException))
             {
-                throw GetCustomException(SR.Format(SR.net_io_readfailure, exception.Message), exception);
-            }
-
-            if (errorCode != SocketError.Success)
-            {
-                var socketException = new SocketException((int)errorCode);
-                throw GetExceptionFromSocketException(SR.Format(SR.net_io_readfailure, socketException.Message), socketException);
+                throw WrapException(SR.net_io_readfailure, exception);
```

Comment:
I assume it was written the way it was written before to save the expense of an extra exception throw/catch.  It's probably not worth the extra code/complexity.

Reply:
If we really want to avoid the extra cost, we probably also want to change other overloads to use the `out` parameter too. (The whole idea started with a discussion about consistency of catch blocks in #40772)